### PR TITLE
fix `addColumn` with reference not add reference in mysql [issues 5592]

### PR DESCRIPTION
**Pull request by ningshen**
- solve issue: [queryInterface.addColumn with references not add reference in mysql](https://github.com/sequelize/sequelize/issues/5592)
- all existing tests pass
- add queryInterface.addColumn test for mysql

### CODE REVIEW

File: lib/dialects/mysql/query-generator.js [hunk 1]
```diff
@@ -265,6 +266,10 @@ var QueryGenerator = {
 
     if (attribute.references) {
       attribute = Utils.formatReferences(attribute);
+      if (options && options.context === 'addColumn' && options.attributeName) {
+        template += ', ADD FOREIGN KEY (' + this.quoteIdentifier(options.attributeName) + ')';
```
Comment (mickhansen) [inline lib/dialects/mysql/query-generator.js]: I haven't checked yet, but does that match the naming we use elsewhere for foreign keys?

File: lib/dialects/mysql/query-generator.js [hunk 2]
```diff
@@ -265,6 +266,16 @@ var QueryGenerator = {
 
     if (attribute.references) {
       attribute = Utils.formatReferences(attribute);
+
+      if (options && options.context === 'addColumn' && options.foreignKey) {
+        var attrName = options.foreignKey;
+
+        template += Utils._.template(', ADD CONSTRAINT <%= fkName %> FOREIGN KEY (<%= attrName %>)')({
+          fkName: this.quoteIdentifier(attrName + '_foreign_idx'),
```
Comment (mickhansen) [inline lib/dialects/mysql/query-generator.js]: Can you point to where you found that naming? It looks like create tables leaves out a name (so it uses the database generated one).

If there's no existing naming please consider making `[table]_[field]_fkey`.

Comment (ningshen) [inline lib/dialects/mysql/query-generator.js]: I follow the ```changeColumnQuery: function(tableName, attributes) ``` with reference, at
lib/dialects/mysql/query-generator.js : Line 131

``` javascript
constraintString.push(Utils._.template('<%= fkName %> FOREIGN KEY (<%= attrName %>) <%= definition %>')({
          fkName: this.quoteIdentifier(attributeName + '_foreign_idx'),
          attrName: this.quoteIdentifier(attributeName),
          definition: definition.replace(/.+?(?=REFERENCES)/,'')
        }));
```

Comment (ningshen) [inline lib/dialects/mysql/query-generator.js]: I follow code at sequelize/test/unit/sql/change-column.test.js: Line 67.

```
expectsql(sql, {
            mssql: 'ALTER TABLE [users] ADD CONSTRAINT [level_id_foreign_idx] FOREIGN KEY ([level_id]) REFERENCES [level] ([id]) ON DELETE CASCADE;',
            mysql: 'ALTER TABLE `users` ADD CONSTRAINT `level_id_foreign_idx` FOREIGN KEY (`level_id`) REFERENCES `level` (`id`) ON DELETE CASCADE ON UPDATE CASCADE;',
            postgres: 'ALTER TABLE "users"  ADD CONSTRAINT "level_id_foreign_idx" FOREIGN KEY ("level_id") REFERENCES "level" ("id") ON DELETE CASCADE ON UPDATE CASCADE;',
          });
```

where set foreign key ```FOREIGN KEY (`level_id`)```

Comment (mickhansen) [inline lib/dialects/mysql/query-generator.js]: Alright, thanks for referencing that for me.

File: test/unit/sql/add-column.test.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      it('properly generate alter queries', function(){
+        return current.getQueryInterface().addColumn(Model.getTableName(), 'level_id', {
+          type: DataTypes.FLOAT,
+          allowNull: false,
+        }).then(function(sql){
+          expectsql(sql, {
+            mysql: 'ALTER TABLE `users` ADD `level_id` FLOAT NOT NULL;',
+          });
+        });
+      });
```
Comment (mickhansen) [inline test/unit/sql/add-column.test.js]: can you please refactor this unit test to not rely on stubbing but simply test the query interface directly?
(goes for both tests really)

**Comment by ningshen** [comment]
It seem My commit can not trigger travis. Do I need pull the latest code from master, then merge and commit and seed pull request again?

**Comment by janmeier** [comment]
@ningshen Your commit cannot be merged cleanly into master right now, that's why travis is not running

```bash
$ git fetch upstream (Or whatever your sequelize/sequelize remote is called)
$ git rebase upstream/master
$git push -f 
```
A force push is needed because rebasing changes existing commits

**Comment by ningshen** [comment]
I make a mistake and delete repository.......
I send another pull request [pr5593](https://github.com/sequelize/sequelize/pull/5593). please check it.

**Pull request closed without merging**